data_steward/tools/create_update_drc_id_match_table.py

```diff
@@ -0,0 +1,229 @@
+"""
+Utility to create or update a site's DRC identity match table.
+
+There should be a record for each participant and the record should be filled with default values of `missing_rdr` or
+    `missing_ehr`. Each record should contain data for the fields: person_id, first_name, middle_name, last_name,
+    phone_number, email, address_1, address_2, city, state, zip, birth_date, sex, and algorithm.
+
+The record for each of the above fields should default to `missing_rdr` if the joined record in the
+    ps_api_values_<hpo_id> table does not contain any information otherwise, it should default to `missing_rdr`
+
+Original Issue: DC-1216
+"""
+
+# Python imports
+import logging
+import argparse
+
+# Project imports
+import resources
+from utils import bq
+from utils import auth
+from bq_utils import table_exists
+from tools.create_tier import SCOPES
+from common import JINJA_ENV, PS_API_VALUES, DRC_OPS
+from constants.validation.participants.identity_match import IDENTITY_MATCH_TABLE
+
+LOGGER = logging.getLogger(__name__)
+
+CREATE_TABLE = JINJA_ENV.from_string("""
+CREATE TABLE `{{project_id}}.{{drc_dataset_id}}.{{id_match_table_id}}` ({{fields}})
+PARTITION BY DATE_TRUNC(_PARTITIONTIME, HOUR)
+""")
+
+INSERT_QUERY = JINJA_ENV.from_string("""
+MERGE `{{project_id}}.{{drc_dataset_id}}.{{id_match_table_id}}` b 
+USING `{{project_id}}.{{drc_dataset_id}}.{{ps_values_table_id}}` a
+ON a.person_id = b.person_id
+WHEN NOT MATCHED THEN 
+INSERT (_PARTITIONTIME, {{fields}}) 
+VALUES (TIMESTAMP_TRUNC(CURRENT_TIMESTAMP, HOUR), {{fields}})
+""")
+
+MISSING_EHR_QUERY = JINJA_ENV.from_string("""
+UPDATE `{{project_id}}.{{drc_dataset_id}}.{{id_match_table_id}}`
+SET {{col}} = 'missing_ehr'
+WHERE {{col}} IS NULL
+""")
+
+MISSING_RDR_QUERY = JINJA_ENV.from_string("""
+UPDATE `{{project_id}}.{{drc_dataset_id}}.{{id_match_table_id}}`
+SET {{set_expression}}
+WHERE COALESCE({{fields}}) IS NULL
```

Comment:
This would set "missing_rdr" only if all fields are null/missing. Isn't the intention to set "missing_rdr" field by field?

Reply:
I thought it was `missing_ehr` for missing information field by field...